source/MissionSharedLibrary/src/Utilities/Utility.cs

```diff
@@ -216,6 +216,12 @@ public static void PlayerControlAgent(Agent agent)
             agent.Formation = null;
             agent.Controller = Agent.ControllerType.Player;
             agent.Formation = formation;
+
+            // Add HumanAIComponent back to agent after player control to avoid crash
+            // when agent dies while climbing ladder
+            // or when trying to control an agent who was using siege weapon
+            agent.AddComponent(new HumanAIComponent(agent));
```

Comment:
Because the AgentHumanAILogic would add HumanAIComponent when AI controls agent, we need to remove this added component when AI controls the main agent to avoid duplicated HumanAIComponent be added to the same agent.

I will approve the changes and add the code afterwards.